internal/pkg/manifest/validate.go

```diff
@@ -509,34 +509,23 @@ func (t TaskConfig) Validate() error {
 
 // Validate returns nil if PlatformArgsOrString is configured correctly.
 func (p PlatformArgsOrString) Validate() error {
-	if p.PlatformString != nil {
-		if err := p.PlatformString.Validate(); err != nil {
-			return err
+	if !p.PlatformArgs.isEmpty() {
```

Comment:
The `platform` field isn'r required right?

If it is not, what do you think we return early if it's empty:

```suggestion
        if p.isEmpty() {
            return nil
        }
        
        if !p.PlatformArgs.isEmpty() {
            // do the thing
        }

      if err := validateShortPlatform(p.PlatformString); err != nil {
           // do the thing
      }
      
      return nil
```